### PR TITLE
add input column to Simbad result, relates to issue #473

### PR DESCRIPTION
**Pull request by gully**
This pull request relates to issue #473.  It adds a column "INPUT" to the result of a Simbad query.  This column is useful when one runs query_objects() with many identifiers, some of which are not recognized by Simbad.  

This PR also adds a flag to query_objects.

I did not make a test for this yet, but I figured I would at least get the ball rolling before I forgot about it, and maybe get some feedback about whether this approach seems OK.  Specifically, we might want to leave it out of query_objects for now, since it is an experimental feature.

As for the test, any advice on where to start is appreciated.  

Once again, any constructive critique is appreciated, this is my first time attempting a PR.

### CODE REVIEW

File: astroquery/simbad/core.py [hunk 1]
```diff
@@ -392,7 +392,47 @@ def query_objects(self, object_names, wildcard=False, verbose=False):
...
+
+        for error in error_string.split("\n"):
+            start_loc = error.rfind(":") + 2
+            fail = error[start_loc:]
+            fails.append(fail)
+
+        successes = [s for s in input_list if s not in fails]
+        if verbose:
+            say = "There were {} successful Simbad matches and {} failures."
+            print say.format(len(successes), len(fails))
```
Comment (keflavich) [inline astroquery/simbad/core.py]: `print` statements are not permitted in astroquery code; instead use `log.info` here.

**Comment by keflavich** [comment]
Besides the above comment (which also identifies the cause of the build failure), the pull request looks good.  For a test case, you can implement a remote one and I'll help or demonstrate how to make a local version.

**Comment by gully** [comment]
Some day I'll close this PR! :)

Just an update, I still find this code snippet super useful.  In fact, I wonder how folks *don't* use this, when they're vetting unknown identifiers in batch mode.  It suggests that other folks have workarounds, or other ways to solve the same task.  I wonder if it's just for loops?  I lost track of this snippet at some point and was resorting to for loops for every object, which was very slow.

**Comment by bsipocz** [comment]
@gully - Please follow the astropy coding guides and rebase rather than merge in the master branch to keep the history clean. Thanks!

**Comment by gully** [comment]
I attempted a rebase, did that work?

**Comment by gully** [comment]
Update--- This PR does not work as it stands:

```python
out = sim.query_objects(vec, keep_input=True)
```
results in:

```python
    597         successes = [s for s in input_list if s not in fails]
    598 
--> 599         self.last_parsed_result.table["INPUT"] = successes

..."could not broadcast input array from shape (2076) into shape (2012)"
```

**Comment by keflavich** [comment]
@gully Is this because failed searches result in no output, as mentioned in #967?

**Comment by gully** [comment]
Sort of? There are different classes of failures, so it's a matter of catching them and extracting substrings:

```python
        error_string = self.last_parsed_result.error_raw
        fails = []
        not_found_str = 'Identifier not found in the database :'
        incorrect_id_str = ': this identifier has'
        no_known_cat = ': No known catalog could be found'
        for error in error_string.split("\n"):
            if error.find(not_found_str) != -1:
                start_loc = error.rfind(not_found_str)+len(not_found_str) + 1
                end_loc = -1
            if error.find(incorrect_id_str) != -1:
                start_loc = error.find("] ") + 3
                end_loc = -error.find("': this id")
            if error.find(no_known_cat) != -1:
                start_loc = error.find("] ") + 3
                end_loc = -error.find("': No known")
            fail = error[start_loc:end_loc]
            fails.append(fail)
            print(fail)
```

**Comment by stargaser** [comment]
@bjfultn was just discussing this feature with me (we share an office). It would be really useful to have the feature of echoing back the user's identifier, and to include rows where Simbad did not return a result.

**Comment by aoberto** [comment]
Hello, now, if you asked for 'typed_id' in the output, in case of unknown object the Simbad result will contain a row with your id and nothing in the others columns. Can you check that it sounds good for you ?

**Comment by bjfultn** [comment]
This solution sounds great, thanks!

**Pull request closed without merging**